lib/messagebird.js

```diff
@@ -395,6 +395,308 @@ module.exports = function (accessKey, timeout) {
           httpRequest('POST', '/lookup/' + phoneNumber + '/hlr', params, callback);
         }
       }
+    },
+
+    contacts: {
+
+      /**
+       * Create a new contact. Params is optional.
+       *
+       * @param {String} phoneNumber
+       * @param {Object} params
+       * @param {Function} callback
+       * @return void
+       */
+      create: function (phoneNumber, params, callback) {
+        if (typeof params === 'function') {
+          callback = params;
+          params = {};
+        }
+
+        params.msisdn = phoneNumber;
+
+        httpRequest('POST', '/contacts', params, callback);
+      },
+
+      /**
+       * Deletes an existing contact. The callback is invoked with an error if
+       * applicable, but the data will never contain anything meaningful as the
+       * API returns an empty response for successful deletes.
+       *
+       * @param {String} id
+       * @param {Function} callback
+       * @return void
+       */
+      delete: function (id, callback) {
+        httpRequest('DELETE', '/contacts/' + id, callback);
+      },
+
+      /**
+       * Lists existing contacts. Pagination is optional. If a limit is set, an
+       * offset is also required.
+       *
+       * @param {Number} limit
+       * @param {Number} offset
+       * @param {Function} callback
+       * @return void
+       */
+      list: function (limit, offset, callback) {
+        var params = null;
+
+        if (typeof callback === 'function') {
+          params = {
+            limit: limit,
+            offset: offset
+          };
+        } else {
+          callback = limit;
+        }
+
+        httpRequest('GET', '/contacts', params, callback);
+      },
+
+      /**
+       * View an existing contact.
+       *
+       * @param {String} id
+       * @param {Function} callback
+       * @return void
+       */
+      read: function (id, callback) {
+        httpRequest('GET', '/contacts/' + id, callback);
+      },
+
+      /**
+       * Updates an existing contact. Params is optional.
+       *
+       * @param {String} id
+       * @param {String} name
+       * @param {Object} params
+       * @param {Function} callback
+       * @return void
+       */
+      update: function (id, params, callback) {
+        if (typeof params === 'function') {
+          callback = params;
+          params = {};
+        }
```

Comment:
This is probably not needed, since this condition is checked by the `httpRequest` method itself.

Reply:
Great catch! I've removed them in https://github.com/messagebird/messagebird-nodejs/pull/20/commits/661cebca7d715e6c5e8b1a9083be0c4cc9025ae1.